File: 力扣/7.整数反转.cpp
/*
 * @lc app=leetcode.cn id=7 lang=cpp
 *
 * [7] 整数反转
 */

// @lc code=start
class Solution
{
public:
    int reverse(int x)
    {
        long long ans = (long long )x;
        long long sum = 0;
        if (x < 0)
        {
            ans = -ans;
        }
        long long  a[40];
        int pos = 0;
        while (ans)
        {
            a[pos++] = (ans % 10);
            ans /= 10;
        }
        for (int i = 0; i <pos; i++)
        {
            sum = sum * 10 + a[i];
            // printf("%lld\n",sum);
        }
        if (sum>INT_MAX){
            return 0;
        }
        if (x < 0)
        {
            return -sum;
        }
        else
        {
            return sum;
        }
    }
};
// @lc code=end

File: 力扣/7.整数反转.py
#
# @lc app=leetcode.cn id=7 lang=python3
#
# [7] 整数反转
#

# @lc code=start
class Solution:
    def reverse(self, x: int) -> int:
        if x==0:
            return 0
        if x>=0:
            x = str(x)
            # print()
            x = x[::-1]    
            for i in range(len(x)):
                if x[i]!='0':
                    return x[i:]
        else:
            x = str(x)
            # print()
            x = f'-{x[::-1][:-1]}'
            # y = x.find('0')
            # print(y)
            # print(x)
            for i in range(1,len(x)):
                # print(x[i])
                if x[i]!='0':
                    return f'-{x[i:]}'
# @lc code=end

